src/components/Navigator/QuickNavigationModal.vue

```diff
@@ -272,6 +343,78 @@ export default {
     startingPointHook() {
       this.focusedIndex = this.totalItemsToNavigate - 1;
     },
+    async fetchSelectedSymbolData() {
+      // start a timer: if a full second has elapsed and the network request for
+      // data hasn't completed yet, indicate that the data is still loading in
+      // the actual UI — without the delay, the constant flashing of this
+      // message would be too distracting
+      this.loadingTimeout = setTimeout(() => {
+        if (this.previewState === PreviewState.loading) {
+          this.previewIsLoadingSlowly = true;
+        }
+      }, SLOW_LOADING_DELAY);
+
+      // exit early if the previwe data is already cached for this symbol as
+      // there is no additional work needed
+      if (!this.selectedSymbol || this.$cachedSymbolResults.has(this.selectedSymbol.uid)) {
+        clearTimeout(this.loadingTimeout);
+        this.previewIsLoadingSlowly = false;
+        return;
+      }
+
+      const fetchSymbolData = async (symbol) => {
+        if (!symbol || this.$cachedSymbolResults.has(symbol.uid)) {
+          return;
+        }
+
+        try {
+          // load render JSON for the selected /documentation/ page
+          const json = await fetchDataForPreview(symbol.path, {
+            signal: this.abortController.signal,
+          });
+          this.$cachedSymbolResults.set(symbol.uid, {
+            success: true,
+            json,
+          });
+        } catch (e) {
+          // errors triggered by the abort controller are safe to ignore since
+          // we are only aborting them for performance reasons and would like
+          // to later re-try them if possible
+          if (e.name !== ABORT_ERROR_NAME) {
+            this.$cachedSymbolResults.set(symbol.uid, {
+              success: false,
+            });
+          }
+        } finally {
+          // `LRUMap` is a custom object that is very similar to a builtin `Map`
+          // and since `Map` objects can't directly hook into the reactivity
+          // in Vue 2, we need to convert this object into a plain `Object` every
+          // time it is mutated to workaround this lack of reactivity
+          this.cachedSymbolResults = Object.fromEntries(this.$cachedSymbolResults);
```

Comment:
The only thing that worries me here is that if the `cachedSymbolResults` becomes full with very very big JSON entries, Vue will try to make them deeply reactive, which for example caused issues in the Navigator.

With the Composition API we could have leveraged `shallowReactive`.

20 max items is not much, so maybe we should just keep an eye out. Easiest solution would be to `Object.freeze` it, if we notice perf issues.

Reply:
Good point, and it technically results in a copy of the full data due to this bit.

I'll see if there's a way around that. Otherwise, I'll look into `Object.freeze`—it sounds like using that would prevent Vue from attaching the getters/setters for the reactive system?

Reply:
Only shallow ones at the top level yes. You dont care about deep changes anyway.

Reply:
I don't think we would ever have 2 back to back entries that are the same, would it make sense to update the `cachedSymbolResults` only when both the current entry and the next entry finished fetching?

Reply:
8beabe4d1e1d57f0067be538fd54adc20473056f